shared/common-adapters/floating-picker.native.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+import React, {type Node} from 'react'
+import {TouchableWithoutFeedback, Picker} from 'react-native'
+import Box, {Box2} from './box'
+import Text from './text'
+import {globalColors, globalMargins, globalStyles} from '../styles'
+import FloatingBox from './floating-box.native'
+import {isAndroid} from '../constants/platform'
+
+type PickerItem = {label: string, value: string | number}
+
+type Props = {
+  items: PickerItem[], // values must be unique
+  selectedValue: string | number,
+  onSelect: (string | number) => void,
+  header?: Node,
+  prompt?: Node,
+  promptString?: string, // used on android as title of selection popup
+  onHidden: () => void,
+  onCancel: () => void,
+  onDone: () => void,
+  visible: boolean,
+}
+
+const FloatingPicker = (props: Props) => {
+  if (!props.visible) {
+    return null
+  }
+  return (
+    // onHidden required by the typing for desktop, not used on mobile
+    <FloatingBox onHidden={() => {}}>
+      <TouchableWithoutFeedback style={styleOverlayContainer} onPress={props.onHidden}>
+        <Box style={styleOverlay}>
+          <Box2 direction="vertical" fullWidth={true} style={styleMenu}>
+            {props.header}
+            <Box2
+              direction="horizontal"
+              gap="small"
+              gapStart={true}
+              gapEnd={true}
+              fullWidth={true}
+              style={styleActionButtons}
+            >
+              <Text
+                type="BodySemibold"
+                style={{color: globalColors.blue, fontSize: 17, flex: 1}}
+                onClick={props.onCancel}
+              >
+                Cancel
+              </Text>
+              <Text
+                type="BodySemibold"
+                style={{color: globalColors.blue, fontSize: 17}}
+                onClick={props.onDone}
+              >
+                Done
+              </Text>
+            </Box2>
+            {props.prompt}
+            <Picker
+              selectedValue={props.selectedValue}
+              onValueChange={(itemValue, itemIndex) => props.onSelect(itemValue)}
+              prompt={props.promptString}
+              style={
+                // Make sure we're far from the action buttons
+                isAndroid ? {marginBottom: globalMargins.large, marginTop: globalMargins.medium} : undefined
```

Comment:
we should get into the habit of using styleSheetCreate and collapseStyles and etc